Libs/Mesh/Mesh.cpp

```diff
@@ -967,6 +989,599 @@ std::ostream& operator<<(std::ostream &os, const Mesh& mesh)
+            vtkSmartPointer<vtkDijkstraGraphGeodesicPath>::New();
+        dijkstra->SetInputData(this->mesh);
+
+        vtkIdType lastId = 0;
+        for(size_t i = 0; i < boundaries[bound].size(); i++){
+            Eigen::Vector3d pt = boundaries[bound][i];
+            double ptdob[3] = {pt[0],pt[1],pt[2]};
+            vtkIdType ptid = locator->FindClosestPoint(ptdob);
+            mesh->GetPoint(ptid, ptdob);
+
+            // Add first point in boundary
+            if(i == 0){
+                lastId = ptid;
+                double pathpt[3];
+                mesh->GetPoint(ptid, pathpt);
+                selectionPoints->InsertNextPoint(pathpt[0],pathpt[1],pathpt[2]);
+                boundaryVerts.push_back(ptid);
+            }
+            // If the current and last vetices are different, then add all vertices in the path to the boundaryVerts list
+            if(lastId != ptid){
+                //std::cout << pt[0] << " " << pt[1] << " " << pt[2] << " -> " << ptdob[0] << " " << ptdob[1] << " " << ptdob[2] << std::endl;
+                // Add points in path
+                dijkstra->SetStartVertex(lastId);
+                dijkstra->SetEndVertex(ptid);
+                dijkstra->Update();
+                vtkSmartPointer<vtkIdList> idL = dijkstra->GetIdList();
+                for(size_t j = 1; j < idL->GetNumberOfIds(); j++){
+                    vtkIdType id = idL->GetId(j);
+                    double pathpt[3];
+                    mesh->GetPoint(id, pathpt);
+                    selectionPoints->InsertNextPoint(pathpt[0],pathpt[1],pathpt[2]);
+                    boundaryVerts.push_back(id);
+                }
+            }
+            lastId = ptid;
+        }
+
+        std::cout << "Number of boundary vertices " << boundaryVerts.size() << std::endl;
+
+        vtkSmartPointer<vtkSelectPolyData> select = vtkSelectPolyData::New();
+        select->SetLoop(selectionPoints);
+        select->SetInputData(this->mesh);
+        select->GenerateSelectionScalarsOn();
+        select->SetSelectionModeToLargestRegion();
+
+        // Clipping mesh
+        vtkSmartPointer<vtkClipPolyData> selectclip = vtkClipPolyData::New();
+        selectclip->SetInputConnection(select->GetOutputPort());
+        selectclip->SetValue(0.0);
+
+        selectclip->Update();
+
+        MeshType halfmesh = selectclip->GetOutput();
+
+        vtkSmartPointer<vtkDoubleArray> inout = computeInOutForFFCs(query, halfmesh);
+
+
+
+        vtkSmartPointer<vtkDoubleArray> values = vtkSmartPointer<vtkDoubleArray>::New();
+        vtkSmartPointer<vtkDoubleArray> absvalues = setDistanceToBoundaryValueFieldForFFCs(values, points, boundaryVerts, inout, V, F, dom);
+
+        this->mesh->GetPointData()->SetActiveScalars("value");
+
+        std::vector<Eigen::Matrix3d> face_grad_ = this->setGradientFieldForFFCs(absvalues, V, F);
+
+        //this->visualizeVectorFieldForFFCs(values, face_grad_, V, F);
```

Comment:
Why is it commented out?

Reply:
Because we want functionality to visualize vector fields, but don't want them to show for a user.

Reply:
Okay. You can add this as a comment